library/core/src/main/java/com/google/android/exoplayer2/audio/AudioCapabilitiesReceiver.java

```diff
@@ -163,4 +186,27 @@ public void onChange(boolean selfChange) {
     }
   }
 
+  // Define the Callback object and its method to detect the Bluetooth speaker connenction
+  private final MediaRouter.Callback mediaRouterCallback =
+          new MediaRouter.Callback() {
+            @Override
+            public void onRouteChanged(MediaRouter router, RouteInfo route) {
+            }
+
+            @Override
+            public void onRouteSelected(MediaRouter router, RouteInfo route) {
```

Comment:
It this called on registration with the current route on registration? If not, then the ExternalPcmOnlySoundset state will be incorrect until the route change.

Reply:
Yes, it is called as a `handleMessage` callback on the `Looper` that calls register if the bluetooth device is connected.  It looks like it may not for HDMI connection, So this looks like a bug.  Good catch, thanks!